### PR TITLE
Open LyberCore::Robot to allow an injected workflow service client

### PR DESCRIPTION
**Pull request by cbeer**
This is the first step to supporting https://github.com/sul-dlss/dor-workflow-service/pull/30, and should allow downstream robots to provide that instantiatable class.

### CODE REVIEW

File: lib/lyber_core/robot.rb [hunk 1]
```diff
@@ -72,21 +72,26 @@ def work(druid)
...
         LyberCore::Log.error "Cannot set #{druid} to status='error'\n" + e2.message + "\n" + e2.backtrace.join("\n")
         raise e2 # send exception to Resque failed queue
       end
     end
 
-  private
+    private
+
+    def workflow_service
+      @workflow_service || Dor::WorkflowService
```
Comment (drh-stanford) [inline lib/lyber_core/robot.rb]: is this meant to be `||=` or is `||` correct?

Comment (cbeer) [inline lib/lyber_core/robot.rb]: I meant `||`. We don't want to hold onto the singleton, and after we have an instantiatable class, I think it should be fairly low-overhead anyway.

**Comment by darrenleeweber** [comment]
It's not immediately obvious what this achieves.  There seems to be no way that `@workflow_service` can be assigned anything.  Where is the associated `attr_?` macro and init code?

**Comment by darrenleeweber** [comment]
:+1: for the init and attr changes, although I'm still kinda :0: on the bigger plan to require instances.

**Comment by darrenleeweber** [comment]
I guess I'm OK with this for merge, but defer to @drh-stanford for more insight on the consequences.

**Pull request merged**